dashboard/src/components/MainView/MainView.js

```diff
@@ -13,16 +13,17 @@ const MainView = () => {
   const [allPollutants, setAllPollutants] = useState();
   const [searchedCity, setSearchedCity] = useState("");
   const [cityData, setCityData] = useState("");
+  const [pollutantID, setPollutantID] = useState();
 
   const onSearchButtonClick = async () => {
```

Comment:
The search box currently doesn't work and requires additional functionalities to be implemented. We want users to be able to search for a specific city or country. When they click the search button, if the API has data for that location, the app should transition to the Detailed View. However, if the API doesn't have data for that location, we need to show a message box to the user, informing them that our website doesn't provide air quality information for that particular location. It might be better to create a separate "Search" component and move all the functionality related to searching into that component.

Reply:
The search functionality in the MainView is currently calling the API and getting the data for the searched for city, we just aren't displaying any data from said search currently. Did we want to display the most recent data measured for the searched for location on the MainView? Or do we want the search functionality within the DetailedView and a modified version in the Historical that accepts dates as well as location?

Reply:
Got it! Here's how we'll proceed:

- On the detailed view, we'll display the most recent data measured for the searched location. Users can directly search for a new location on the detailed view itself without needing to go back to the main view.
- From the detailed view, users can navigate to the historical view to see the air quality data for the searched location within a specific date range. However, there won't be a search box on the historical view to enter a location.
- In the future, we'll remove the navigation bar from the main view, so the only way to access the detailed view and historical view will be by searching for a location.

Hope that clears things up.

Reply:
Okay, I already began working on tying in data for the detailed view, it should be pretty easy to update with a search being passed to it from the MainView.

Reply:
Do we want the dropdown I made to select city or country that is currently in the DetailedView to be available in the MainView?

Reply:
In my opinion, having it is not necessary, but if you feel it's essential for the search functionality, you can keep it there. However, it's important to ensure that the searching process remains consistent between the main and detailed views.

Reply:
I agree, which is why I want to pull it out as a component and have it act consistently between the two.